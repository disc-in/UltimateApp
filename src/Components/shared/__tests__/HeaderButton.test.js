import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import HeaderButton from '../HeaderButton';

describe('<HeaderButton />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<HeaderButton icon="check" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls delegates onPress prop', async () => {
    const onPress = jest.fn();

    const { getByTestId } = render(<HeaderButton onPress={onPress} />);

    await fireEvent.press(getByTestId('headerButton'));

    expect(onPress).toHaveBeenCalled();
  });
});
