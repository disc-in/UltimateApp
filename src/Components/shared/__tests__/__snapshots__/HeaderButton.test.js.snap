// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="headerButton"
>
  <Text />
</View>
`;
